lib/ret.ex

```diff
@@ -1,17 +1,27 @@
 defmodule Ret do
   import Canada, only: [can?: 2]
   import Ecto.Query
-  alias Ret.{Account, Avatar, AvatarListing, OwnedFile, Repo, Storage}
+  alias Ret.{Account, Asset, Avatar, AvatarListing, OwnedFile, Project, Repo, Storage}
 
   def get_account_by_id(account_id) do
     Repo.get(Account, account_id)
   end
 
+  @asset_file_columns [:asset_owned_file, :thumbnail_owned_file]
+  @project_file_columns [:project_owned_file, :thumbnail_owned_file]
   def delete_account(%Account{} = acting_account, %Account{} = account_to_delete) do
     if can?(acting_account, delete_account(account_to_delete)) do
       reassign_avatar_listings(account_to_delete, acting_account)
       reassign_parent_avatars(account_to_delete, acting_account)
-      delete_avatars_for_account(account_to_delete)
+
+      for {query, file_columns} <- [
+            {from(avatar in Avatar, where: avatar.account_id == ^account_to_delete.account_id), Avatar.file_columns()},
+            {from(asset in Asset, where: asset.account_id == ^account_to_delete.account_id), @asset_file_columns},
+            {from(project in Project, where: project.created_by_account_id == ^account_to_delete.account_id),
+             @project_file_columns}
+          ] do
+        delete_entities_with_owned_files(query, file_columns)
+      end
```

Comment:
Since this is just sending the double to the function as arguments, this might be easier to grok without the list comprehension.

```elixir
delete_entities_with_owned_files(
  from(avatar in Avatar, where: avatar.account_id == ^account_to_delete.account_id),
  Avatar.file_columns()
)

delete_entities_with_owned_files(
  from(asset in Asset, where: asset.account_id == ^account_to_delete.account_id),
  @asset_file_columns
)

delete_entities_with_owned_files(
  from(project in Project, where: project.created_by_account_id == ^account_to_delete.account_id),
  @project_file_columns
)
```

It repeats characters (i.e., `delete_entities_with_owned_files`) but not logic.